mat64/eigen.go

```diff
@@ -798,22 +800,23 @@ func hqr2(d, e []float64, hess, v *Dense, epsilon float64) {
 	}
 }
 
-// D returns the block diagonal eigenvalue matrix from the real and imaginary
-// components d and e.
-func (f EigenFactors) D() *Dense {
-	d, e := f.d, f.e
-	var n int
-	if n = len(d); n != len(e) {
-		panic(ErrSquare)
-	}
-	dm := NewDense(n, n, nil)
-	for i := 0; i < n; i++ {
-		dm.set(i, i, d[i])
-		if e[i] > 0 {
-			dm.set(i, i+1, e[i])
-		} else if e[i] < 0 {
-			dm.set(i, i-1, e[i])
+type EigenValues []complex128
+
+// D returns the eigenvalue matrix from the real and imaginary components
+// of the eigenvalues.
+//
+// If the decomposed matrix is not symmetric, then D returns a block diagonal
+// matrix with the real eigenvalues in 1-by-1 blocks and any complex eigenvalues,
+// lambda + i*mu, in 2-by-2 blocks, [lambda, mu; -mu, lambda].
+func (v EigenValues) D() *Dense {
```

Comment:
I use octave, so that might explain the difference.

>The bigger issue is that the mat64 eigenvectors are not orthonormal, and not only that, appear to not even be eigenvalues

Yes, that is the issue that I am trying to sort out.